src/lib/expression/evaluation/expression_evaluator.hpp

```diff
@@ -0,0 +1,145 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "boost/variant.hpp"
+
+#include "types.hpp"
+#include "all_type_variant.hpp"
+#include "null_value.hpp"
+#include "expression/logical_expression.hpp"
+#include "expression_result.hpp"
+
+namespace opossum {
+
+class AbstractExpression;
+class ArithmeticExpression;
+class BaseColumn;
+class BinaryPredicateExpression;
+class CaseExpression;
+class Chunk;
+class ExistsExpression;
+class ExtractExpression;
+class FunctionExpression;
+class InExpression;
+class PQPSelectExpression;
+
+struct BaseColumnMaterialization {
+  virtual ~BaseColumnMaterialization() = default;
+};
+
+template<typename T>
+struct ColumnMaterialization : public BaseColumnMaterialization {
```

Comment:
I feel like we have this in a lot of places.
(a) how is this different from a ValueColumn?
(b) if it is sufficiently different, we should have one single definition of this that is also used by the different operators (sort? aggregate?)